### ISSUE TITLE
View button goes to "about:blank"

### ISSUE THREAD

**Issue by hcanning2014** [modxcms/revolution#12250]
Revo 2.3, PHP Version 5.5.9-1ubuntu4.5, Ubuntu 14.04LTS, MySQL 5.5.37

see also the same bug: http://tracker.modx.com/issues/7481

Hi Guys,
I have a server set up with Linode with Revo 2.3, PHP Version 5.5.9-1ubuntu4.5, Ubuntu 14.04LTS, MySQL 5.5.37, Apache 2.2 and all the requirements at http://rtfm.modx.com/revolution/2.x/getting-started/server-requirements
I have the Articles extra installed and if I create an article or resource, save it and click the View button; on window reload I get about:blank in the browser instead of the correct url. If I try and batch delete multiple articles they delete but don't disappear as they should in the listing. Im not sure if these are ajax or JS or suhosin issues because the same website works on my other modxcloud server. It must be some setting I've missed. Is it maybe a permission issue?
Can anyone recommend any settings I should try?
THANKS!

**Comment by gadgetto** [comment]
I can't reproduce this problem.
Also you should post this in the Articles repo as this is an Articles related problem.

**Comment by hcanning2014** [comment]
I host the exact website on modxcloud.com and Linode and the articles extra
works fine. Listings all appear. Some css or js compression issue or
suhosin or curl maybe. Who knows but I just need things to work. Thanks

**Comment by hcanning2014** [comment]
Search the linode reference. Articles works fine on modxcloud.com hosting.

**Comment by hcanning2014** [comment]
Scratch not search. Damn phone changing my word. :)

**Comment by pixelchutes** [comment]
I have run into something similar myself on a few occasions, and it seems to be a side effect if PHP OPcache is enabled. For example, running a setup using PHP 5.3.29 and [ZendOpcache 7.0.3](http://pecl.php.net/package/ZendOpcache) (or potentially any PHP 5.5+ builds), you may be able to reproduce this behavior by simply:

1. Creating a new Resource.
2. Clicking "Save".
3. Quickly clicking "View" after the refresh.

Say your `opcache.revalidate_freq = 60` seconds, the "about:blank" behavior described above will occur for ~1 minute, or until manually flushing / resetting the opcache: 
```php
opcache_reset();
```

I cannot speak to the Articles relationship, but I am able to reproduce the described behavior within MODX core.

**Comment by hcanning2014** [comment]
Turns out having APC installed on the server fixes my Articles issues.

**Comment by rtripault** [comment]
Guys, was the "Empty cache" checkbox checked when you created the resource ?

**Comment by pixelchutes** [comment]
Yes.

**Comment by rtripault** [comment]
OK, was asking because i was able to produce using PHP 5.6.4 & Zned OPcache v7.0.4.

My issue was that "Empty cache" was not checked, resulting in `modContext.aliasMap` not being refresh (while `cache_alias_map` setting was active).
That's basically the way it should "behaves", and the issue was more like a logic problem when being on the manager.
So that sounds to be another issue than the one you've been experiencing.

Anyway, would you mind setting `log_level` to `3`, "reproduce" the issue, and check your error log to see if anything arises ?
Thanks

**Comment by pixelchutes** [comment]
@rtripault, here you go:

```
[2015-01-30 15:42:06] (WARN @ /connectors/index.php) `123` was requested but no alias was located.
[2015-01-30 15:42:08] (WARN @ /manager/index.php) `123` was requested but no alias was located.
[2015-01-30 15:42:09] (WARN @ /connectors/index.php) `123` was requested but no alias was located.
```

**Comment by rtripault** [comment]
OK, this appears to be coming from [modContext.getResourceURI](https://github.com/modxcms/revolution/blob/master/core/model/modx/modcontext.class.php#L398) (see [here](https://github.com/modxcms/revolution/blob/master/core/model/modx/modcontext.class.php#L244)), which did not find the uri in the aliasMap. That means the context settings (which also hold the aliasMap) might not be refreshed (hence my question about the "empty cache" checkbox).

That's how far i've been able to track  the issue. Now it's your turn to track why the aliasMap is not populated/refreshed -- you might also want to confirm that's indeed the root of the issue, by disabling aliasMap use)

Hope that will help.

**Comment by pixelchutes** [comment]
Ok, I'll dig in and see what I can find. I can definitely confirm an empty `preview_url: ""`

**Comment by pixelchutes** [comment]
@rtripault do you see any issue with this approach? I can also confirm the issue is reproducible on my end using Zend Opcache 7.0.4

**Comment by opengeek** [comment]
Why is Zend OpCache keeping stale copies of files when they get updated? That sounds like a bug in Zend OpCache to me. I've never ever had to call a method to deal with the APC opcode cache, and I'm curious why we would have to now....

**Comment by pixelchutes** [comment]
Not sure either, one thing I've noticed is that whenever this occurs, clearing MODX cache is the only way to resolve, as it has cached aforementioned stale copy. `opcache_reset()` has no effect unless ran _before_ MODX cache refresh occurs, and `opcache.revalidate_freq` value is irrelevant at that point.

**Comment by opengeek** [comment]
I wonder if this has something to do with how `xPDOCacheHandler::writeFile()` actually writes the file? Hmm...

**Comment by pixelchutes** [comment]
I believe this has been resolved by #12306 ...I can no longer reproduce the `about:blank` behavior when clicking the preview button with the new changes merged.
